nwss_wastewater/delphi_nwss/run.py

```diff
@@ -0,0 +1,164 @@
+
+
+def generate_weights(df, column_aggregating="pcr_conc_smoothed"):
+    """
+    Weigh column_aggregating by population.
+
+    generate the relevant population amounts, and create a weighted but
+    unnormalized column, derived from `column_aggregating`
+    """
+    # set the weight of places with na's to zero
+    df[f"relevant_pop_{column_aggregating}"] = (
+        df["population_served"] * df[column_aggregating].notna()
+    )
+    # generate the weighted version
+    df[f"weighted_{column_aggregating}"] = (
+        df[column_aggregating] * df[f"relevant_pop_{column_aggregating}"]
+    )
+    return df
+
+
+def weighted_state_sum(df: pd.DataFrame, geo: str, sensor: str):
+    """Sum sensor, weighted by population for non NA's, grouped by state."""
+    agg_df = df.groupby(["timestamp", geo]).agg(
+        {f"relevant_pop_{sensor}": "sum", f"weighted_{sensor}": sum_all_nan}
+    )
+    agg_df["val"] = agg_df[f"weighted_{sensor}"] / agg_df[f"relevant_pop_{sensor}"]
+    agg_df = agg_df.reset_index()
+    agg_df = agg_df.rename(columns={"state": "geo_id"})
+    return agg_df
+
+
+def weighted_nation_sum(df: pd.DataFrame, sensor: str):
+    """Sum sensor, weighted by population for non NA's."""
+    agg_df = df.groupby("timestamp").agg(
+        {f"relevant_pop_{sensor}": "sum", f"weighted_{sensor}": sum_all_nan}
+    )
+    agg_df["val"] = agg_df[f"weighted_{sensor}"] / agg_df[f"relevant_pop_{sensor}"]
+    agg_df = agg_df.reset_index()
+    agg_df["geo_id"] = "us"
+    return agg_df
+
+
+def add_needed_columns(df, col_names=None):
+    """Short util to add expected columns not found in the dataset."""
+    if col_names is None:
+        col_names = ["se", "sample_size"]
+
+    for col_name in col_names:
+        df[col_name] = np.nan
+    df = add_default_nancodes(df)
+    return df
+
+
+def logging(start_time, run_stats, logger):
+    """Boilerplate making logs."""
+    elapsed_time_in_seconds = round(time.time() - start_time, 2)
+    min_max_date = run_stats and min(s[0] for s in run_stats)
+    csv_export_count = sum(s[-1] for s in run_stats)
+    max_lag_in_days = min_max_date and (datetime.now() - min_max_date).days
+    formatted_min_max_date = min_max_date and min_max_date.strftime("%Y-%m-%d")
+    logger.info(
+        "Completed indicator run",
+        elapsed_time_in_seconds=elapsed_time_in_seconds,
+        csv_export_count=csv_export_count,
+        max_lag_in_days=max_lag_in_days,
+        oldest_final_export_date=formatted_min_max_date,
+    )
+
+
+def run_module(params):
+    """
+    Run the indicator.
+
+    Arguments
+    --------
+    params:  Dict[str, Any]
+        Nested dictionary of parameters.
+    """
+    start_time = time.time()
+    logger = get_structured_logger(
+        __name__,
+        filename=params["common"].get("log_filename"),
+        log_exceptions=params["common"].get("log_exceptions", True),
+    )
+    export_dir = params["common"]["export_dir"]
+    token = params["indicator"]["token"]
```

Comment:
can we call this something more descriptive, at least in the parameters file where it might be seen without much context?
```suggestion
    token = params["indicator"]["socrata_token"]
```